Cassette/CASQueueFile.m

```diff
@@ -107,12 +107,29 @@ + (void)safelySetUpFile:(NSString *)path forManager:(NSFileManager *)fileManager
     NSError *tapeError;
 
     // Use a temporary file so we don't leave a partially-initialized file.
-    NSString *tempPath = [NSString stringWithFormat:@"%@.tmp", path];
+    NSString *tempPath = [path stringByAppendingPathExtension:@"tmp"];
 
     // Write the initial set of data for the file
     NSMutableData *fileBuffer = [NSMutableData dataWithLength:QueueFileInitialLength];
     writeInt(fileBuffer, 0, QueueFileInitialLength);
 
+    BOOL isDirectory = YES;
+
+    NSString *folderPath = [path stringByDeletingLastPathComponent];
+
+    BOOL doesFolderExistAlready = [fileManager fileExistsAtPath:folderPath
+                                                    isDirectory:&isDirectory];
+
+    if (!doesFolderExistAlready) {
+        tapeError = [CASError createError:CASErrorFileInitialization];
+        CASLOG(@"Creation of intermediary directories is not support, please create folder path: %@", folderPath);
+    }
+
+    if (doesFolderExistAlready && !isDirectory) {
```

Comment:
I wonder if its better to just handle the error and return in these statements

Reply:
actually I think it would be better to adjust this to just return and remove the CAError handler, which would otherwise add more unnecessary lines of code. Makes more sense to me to write to the error directly instead of creating a new one to avoid writing to this error directly.
- Actually, I think we should re-evaluate using `CAError errorHandler` altogether, things seem less clear to me everywhere else because of it.

Reply:
re-added CASError `handleError` as after discussing it turns out it handles an edge case condition where error in checking error value.